src/tbb/governor.cpp

```diff
@@ -192,6 +192,22 @@ static std::uintptr_t get_stack_base(std::size_t stack_size) {
 #endif /* __TBB_USE_WINAPI */
 }
 
+static std::size_t get_stack_size(arena &a) {
+#if __linux__ && !__bg__
+    size_t np_stack_size = 0;
+    pthread_attr_t np_attr_stack;
+    if (0 == pthread_getattr_np(pthread_self(), &np_attr_stack)) {
+        if (0 == pthread_attr_getstacksize(&np_attr_stack, &np_stack_size)) {
+            __TBB_ASSERT( np_stack_size > 0, "stack size must be positive" );
+        }
+        pthread_attr_destroy(&np_attr_stack);
+    }
+    return np_stack_size > 0 ? np_stack_size : a.my_threading_control->worker_stack_size();
+#else
+    return a.my_threading_control->worker_stack_size();
+#endif
```

Comment:
Can you please add implementation for the rest of OS? (The same logic as for `get_stack_base`)

Reply:
The rest of OS are no portable way to get stack size.
get_stack_base() like the below.
- windows: get stack_base and stack_limit (no stack size)
- linux: pthread_attr_np API
- others: no way for stack base and size -> just using current stack pointer